tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -323,6 +323,151 @@ TEST_F( EnergyPlusFixture, SupervisoryControl_Component ) {
+		EXPECT_EQ(Node( 2 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateRequest, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRate, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateRequest, NodeMdot );
+	
+		PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).EMSLoadOverrideValue = 1.0;
+
+		EMSManager::ManageEMS( DataGlobals::emsCallFromBeginNewEvironmentAfterWarmUp, anyRan );
+
+		EXPECT_FALSE(PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).EMSLoadOverrideOn );
+		EXPECT_NEAR(PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).EMSLoadOverrideValue, 1.0, 0.000001 );
+		SetComponentFlowRate( NodeMdot, 1, 2, 1, 1, 1, 1 );
+		EXPECT_EQ(Node( 1 ).MassFlowRate, NodeMdot );
+		EXPECT_EQ(Node( 1 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 1 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 1 ).MassFlowRateRequest, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRate, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateRequest, NodeMdot );
+		SetComponentFlowRate( NodeMdot, 2, 3, 1, 1, 1, 1 );
+		EXPECT_EQ(Node( 2 ).MassFlowRate, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateRequest, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRate, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateRequest, NodeMdot );
+
+		EMSManager::ManageEMS( DataGlobals::emsCallFromBeginTimestepBeforePredictor, anyRan );
+
+		EXPECT_TRUE(PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).EMSLoadOverrideOn );
+		EXPECT_NEAR(PlantLoop( 1 ).LoopSide( 1 ).Branch( 1 ).Comp( 1 ).EMSLoadOverrideValue, 0.0, 0.000001 );
+		Real64 tempNodeMdot( NodeMdot );
+		SetComponentFlowRate( tempNodeMdot, 1, 2, 1, 1, 1, 1 );
+		EXPECT_EQ(Node( 1 ).MassFlowRate, 0.0 );
+		EXPECT_EQ(Node( 1 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 1 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 1 ).MassFlowRateRequest, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRate, 0.0 );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateRequest, NodeMdot );
+		tempNodeMdot = NodeMdot;
+		SetComponentFlowRate( tempNodeMdot, 2, 3, 1, 1, 1, 1 );
+		EXPECT_EQ(Node( 2 ).MassFlowRate, 0.0 );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 2 ).MassFlowRateRequest, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRate, 0.0 );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMax, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateMaxAvail, NodeMdot );
+		EXPECT_EQ(Node( 3 ).MassFlowRateRequest, NodeMdot );
+
+}
+
```

Comment:
I really like this unit test, really cool blend of idf input and building out a simple plant loop data structure.  

The only comment i have is that I think now we want all new unit tests to have a brief set of comments with some sort of narrative describing what the test is trying to accomplish.